packages/panzer/adapters-ioss/src/Panzer_IOSSConnManager.cpp

```diff
@@ -40,21 +40,970 @@
+	    }
+	    if (dim > 1) {
+	      for (int edge = 0; edge < edgesPerElement; ++edge) {
+	        numCornerNodesThisSubcell = numCornerNodesOnEdges[edge];
+	        TEUCHOS_TEST_FOR_EXCEPTION(numCornerNodesThisSubcell > MAX_SUBCELL_CORNER_NODES_, std::logic_error,
+	                    "Error, currently IOSSConnManager only supports element edges with at most " << MAX_SUBCELL_CORNER_NODES_ << " corner nodes.");
+	        //cornerNodesThisSubcell.resize(numCornerNodesThisSubcell);
+	        for (int cornerNode = 0; cornerNode < numCornerNodesThisSubcell; ++cornerNode) {
+	          cornerNodesThisSubcell[cornerNode] = nodesThisElement[ edgeConnectivities[edge][cornerNode] ];
+	        }
+	        for (int cornerNode = numCornerNodesThisSubcell; cornerNode < MAX_SUBCELL_CORNER_NODES_; ++cornerNode) {
+	          cornerNodesThisSubcell[cornerNode] = -1;
+	        }
+	        // Sort is necessary to ensure edge identifiers are unique.
+	        std::sort(cornerNodesThisSubcell.begin(), cornerNodesThisSubcell.begin()+numCornerNodesThisSubcell);
+	        edgeNodeToEdgeMap_[cornerNodesThisSubcell] = -1;
+	        edgesThisElement.push_back(cornerNodesThisSubcell);
+	      }
+	      if (dim > 2) {
+	        for (int face = 0; face < facesPerElement; ++face) {
+	    	  numCornerNodesThisSubcell = numCornerNodesOnFaces[face];
+	    	  TEUCHOS_TEST_FOR_EXCEPTION(numCornerNodesThisSubcell > MAX_SUBCELL_CORNER_NODES_, std::logic_error,
+	    	  	                    "Error, currently IOSSConnManager only supports element faces with at most " << MAX_SUBCELL_CORNER_NODES_ << " corner nodes.");
+	    	  //cornerNodesThisSubcell.resize(numCornerNodesThisSubcell);
+	    	  for (int cornerNode = 0; cornerNode < numCornerNodesThisSubcell; ++cornerNode) {
+	    	    cornerNodesThisSubcell[cornerNode] = nodesThisElement[ faceConnectivities[face][cornerNode] ];
+	    	  }
+	    	  for (int cornerNode = numCornerNodesThisSubcell; cornerNode < MAX_SUBCELL_CORNER_NODES_; ++cornerNode) {
+	    	    cornerNodesThisSubcell[cornerNode] = -1;
+	    	  }
+	    	  // Sort is necessary to ensure edge identifiers are unique.
+	    	  std::sort(cornerNodesThisSubcell.begin(), cornerNodesThisSubcell.begin()+numCornerNodesThisSubcell);
+	    	  faceNodeToFaceMap_[cornerNodesThisSubcell] = -1;
+	    	  facesThisElement.push_back(cornerNodesThisSubcell);
+	    	}
+	      }
+	    }
+	    elementEdgeNodes_[elementBlockId]->push_back(edgesThisElement);
+	    elementFaceNodes_[elementBlockId]->push_back(facesThisElement);
+	  }
+	}
+  }
+
+  dim = iossElementBlockTopologies_.find(elementBlockIds[0])->second->spatial_dimension();
+
+  std::vector<std::array<GlobalOrdinal,MAX_SUBCELL_CORNER_NODES_>> keys;
+  std::vector<GlobalOrdinal> gids;
+  GlobalOrdinal numEdges = 0;
+  GlobalOrdinal numFaces = 0;
+
+  if (dim > 1) {
+    numEdges = edgeNodeToEdgeMap_.size();
+    if (dim > 2) {
+      numFaces = faceNodeToFaceMap_.size();
+    }
+  }
+
+#ifdef HAVE_MPI
```

Comment:
You should be able to avoid the `ifdef / endif` here for everything but the actual communicator creation, just with a run-time `if (comm.getSize() > 1)`.  That might give you an optimization for the single process case.

The fewer ifdefs you have, the easier it is to avoid build errors and bugs.